source/system/structs/List.ooc

```diff
@@ -21,6 +21,19 @@ List: abstract class <T> {
 	clear: abstract func
 	reverse: abstract func -> This<T>
 	search: abstract func (matches: Func (T) -> Bool) -> Int
+	exists: func (predicate: Func (T) -> Bool) -> Bool {
+		this search(predicate) > -1
+	}
+	all: func (predicate: Func (T) -> Bool) -> Bool {
+		result := true
+		for (index in 0 .. this count)
+			if (!predicate(this[index])) {
+				result = false
+				break
+			}
+		(predicate as Closure) free(Owner Receiver)
+		result
+	}
```

Comment:
I'd prefer these functions at the bottom, so the list of all abstract functions is neat and unbroken.